type-definitions/flow-tests/record.js

```diff
@@ -146,14 +144,15 @@ const mistakeNewInstance = new MakePointNew({ x: 'string' });
 // Note use of + for Read Only.
 type TPerson = { +name: string, +age: number };
 const defaultValues: TPerson = { name: 'Aristotle', age: 2400 };
-const PersonRecord = Record(defaultValues);
+const PersonRecord = Record<TPerson>(defaultValues);
 
-class Person extends PersonRecord<TPerson> {
+class Person extends PersonRecord {
   getName(): string {
     return this.get('name');
   }
 
   setName(name: string): this & TPerson {
+    // $FlowIssue[incompatible-return]
     return this.set('name', name);
   }
```

Comment:
With the new changes, Flow doesn't like the `this & TPerson` return type one bit:

```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ type-definitions/flow-tests/record.js:155:12

Cannot return this.set(...) because: [incompatible-return]
 • Either RecordInstance [1] is incompatible with this [2].
 • Or TPerson [3] is incompatible with this [2].

     type-definitions/flow-tests/record.js
 [3]  147│ const PersonRecord = Record<TPerson>(defaultValues);
         :
      152│   }
      153│
 [2]  154│   setName(name: string): this & TPerson {
      155│     return this.set('name', name);
      156│   }
      157│ }
      158│

     type-definitions/immutable.js.flow
 [1] 1659│   set<K: $Keys<T>>(key: K, value: $ElementType<T, K>): this & $ReadOnly<T>;



Found 1 error
```